lib/azure/service_management/ASM_interface.rb

```diff
@@ -0,0 +1,56 @@
+#
+# Author:: Barry Davis (barryd@jetstreamsoftware.com)
+# Copyright:: Copyright (c) 2010-2011 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'azure/azure_interface'
+require 'azure/service_management/rest'
+require 'azure/service_management/connection'
+
+class Azure
+  class ServiceManagement
+    class ASMInterface < AzureInterface
+      include AzureAPI
+
+      attr_accessor :connection
+
+      def initialize(params = {})
+        @rest = Rest.new(params)
+        @connection = Azure::ServiceManagement::Connection.new(@rest)

```

Comment:
Here we are passing `Rest` object to `connection`. If we don't come up with a situation that requires `@rest` object to be used in the `ASMInterface` itself, we can move `Rest` initialization to `connection`.